### PR TITLE
python3Packages.torchvision: 0.14.1 -> 0.15.1

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

Update `torchvision` to the latest version

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@SomeoneSerge @ericsagnes

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 227952` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.pywick</li>
    <li>python310Packages.pywick.dist</li>
    <li>python310Packages.timm</li>
    <li>python310Packages.timm.dist</li>
  </ul>
</details>
<details>
  <summary>40 packages built:</summary>
  <ul>
    <li>easyocr (python310Packages.easyocr)</li>
    <li>easyocr.dist (python310Packages.easyocr.dist)</li>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>larynx-train (python310Packages.larynx-train)</li>
    <li>larynx-train.dist (python310Packages.larynx-train.dist)</li>
    <li>python310Packages.boxx</li>
    <li>python310Packages.boxx.dist</li>
    <li>python310Packages.bpycv</li>
    <li>python310Packages.bpycv.dist</li>
    <li>python310Packages.clip</li>
    <li>python310Packages.clip.dist</li>
    <li>python310Packages.fastai</li>
    <li>python310Packages.fastai.dist</li>
    <li>python310Packages.ffcv</li>
    <li>python310Packages.ffcv.dist</li>
    <li>python310Packages.grad-cam</li>
    <li>python310Packages.grad-cam.dist</li>
    <li>python310Packages.pytorch-lightning</li>
    <li>python310Packages.pytorch-lightning.dist</li>
    <li>python310Packages.pytorch-metric-learning</li>
    <li>python310Packages.pytorch-metric-learning.dist</li>
    <li>python310Packages.pytorch-pfn-extras</li>
    <li>python310Packages.pytorch-pfn-extras.dist</li>
    <li>python310Packages.sentence-transformers</li>
    <li>python310Packages.sentence-transformers.dist</li>
    <li>python310Packages.tensorboardx</li>
    <li>python310Packages.tensorboardx.dist</li>
    <li>python310Packages.torch-tb-profiler</li>
    <li>python310Packages.torch-tb-profiler.dist</li>
    <li>python310Packages.torchinfo</li>
    <li>python310Packages.torchinfo.dist</li>
    <li>python310Packages.torchvision</li>
    <li>python310Packages.torchvision.dist</li>
    <li>python310Packages.trainer</li>
    <li>python310Packages.trainer.dist</li>
    <li>python310Packages.zcs</li>
    <li>python310Packages.zcs.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
All of the failed builds are also failing on `master`. So I guess that this update is fine.

**Review by samuela — CHANGES_REQUESTED** [review]
don't forget to update the hash on line 52!

**Comment by GaetanLepage** [comment]
> don't forget to update the hash on line 52!

Indeed ! Fixed

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 227952` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.pywick</li>
    <li>python310Packages.pywick.dist</li>
    <li>python310Packages.timm</li>
    <li>python310Packages.timm.dist</li>
  </ul>
</details>
<details>
  <summary>40 packages built:</summary>
  <ul>
    <li>easyocr (python310Packages.easyocr)</li>
    <li>easyocr.dist (python310Packages.easyocr.dist)</li>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>larynx-train (python310Packages.larynx-train)</li>
    <li>larynx-train.dist (python310Packages.larynx-train.dist)</li>
    <li>python310Packages.boxx</li>
    <li>python310Packages.boxx.dist</li>
    <li>python310Packages.bpycv</li>
    <li>python310Packages.bpycv.dist</li>
    <li>python310Packages.clip</li>
    <li>python310Packages.clip.dist</li>
    <li>python310Packages.fastai</li>
    <li>python310Packages.fastai.dist</li>
    <li>python310Packages.ffcv</li>
    <li>python310Packages.ffcv.dist</li>
    <li>python310Packages.grad-cam</li>
    <li>python310Packages.grad-cam.dist</li>
    <li>python310Packages.pytorch-lightning</li>
    <li>python310Packages.pytorch-lightning.dist</li>
    <li>python310Packages.pytorch-metric-learning</li>
    <li>python310Packages.pytorch-metric-learning.dist</li>
    <li>python310Packages.pytorch-pfn-extras</li>
    <li>python310Packages.pytorch-pfn-extras.dist</li>
    <li>python310Packages.sentence-transformers</li>
    <li>python310Packages.sentence-transformers.dist</li>
    <li>python310Packages.tensorboardx</li>
    <li>python310Packages.tensorboardx.dist</li>
    <li>python310Packages.torch-tb-profiler</li>
    <li>python310Packages.torch-tb-profiler.dist</li>
    <li>python310Packages.torchinfo</li>
    <li>python310Packages.torchinfo.dist</li>
    <li>python310Packages.torchvision</li>
    <li>python310Packages.torchvision.dist</li>
    <li>python310Packages.trainer</li>
    <li>python310Packages.trainer.dist</li>
    <li>python310Packages.zcs</li>
    <li>python310Packages.zcs.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>

**Review by samuela — APPROVED** [review]
LGTM all these failures are identical to the ones before so nothing's broken

**Comment by samuela** [comment]
Thanks @GaetanLepage !
